### DISCUSSION TITLE
Why is tab Id different for different file formats?

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1040] [Q&A] [answered]
Hi,
I use playwright to test my application, I open a simple text, and the tab selector is as follows
“shell-tab-code-editor-opener:file:///c%3A/Users/hp/AppData/Local/Temp/cloud-ws-93AtDa/sample.txt:1”
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/27d6548a-432c-4e8e-b056-19524e0a7449)
I open my custom file. the tab selector is as follows
“shell-tab-vdm-diagram:file:///c:/Users/hp/AppData/Local/Temp/cloud-ws-93AtDa/example.vdm”
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/cc793f72-c435-40ff-aef8-50a063a5f670)
the difference is that the simple.txt use “c%3A” but my file use “c:”, why does this difference exist, I think the tab id is generated by Theia, it should be the same for every tab id.
because of this difference, I can not use the same method for both file formats
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/9b0f50d9-6dab-4129-8c74-314913795f88)

image
image
1491×557 50 KB

image

**Comment by wss29** [comment, marked as answer]
Hi,
my vdm format in fact is a glsp diagram, so the widget id is created by the glsp manager. I debug and find that it uses uri.Tostring(true) which translates “c%3A” to “c:”.
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/e56932be-a846-49c1-925d-373eb106d60f)
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/48a747fe-a5a5-41c2-bab6-68e94f3afa46)
this makes glsp diagram file tab id is different from other file tab id in theia.

**Reply by planger** [reply]
You are right, this is what causes the difference. I agree that this difference to the Theia text editor may be somewhat cumbersome. Please feel free to open a PR that aligns the ID for GLSP editors with the ones of Theia text editor.
